### PR TITLE
APIGOV-26695 - update the handling and creation of api transactions for the kong agent

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

**Review by sbolosan — APPROVED** [review]

File: pkg/traceability/processor/handler.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
...0)\",\"via\":\"kong/3.4.1.0-enterprise-edition\",\"x-kong-proxy-latency\":\"9\",\"x-kong-upstream-latency\":\"238\"},\"status\":500,\"size\":422},\"route\":{\"id\":\"c63d3f96-e178-454f-8b70-4efc9bfef5e9\",\"updated_at\":1698677184,\"protocols\":[\"http\",\"https\"],\"strip_path\":true,\"created_at\":1698333843,\"ws_id\":\"973db277-ffd7-4db0-a837-d35edc3690cc\",\"service\":{\"id\":\"069c7945-07ae-4197-a6cb-a5c1ae968d73\"},\"name\":\"PetStoreRoot\",\"hosts\":null,\"preserve_host\":false,\"regex_priority\":0,\"paths\":[\"/petstore3\"],\"response_buffering\":true,\"https_redirect_status_code\":426,\"path_handling\":\"v0\",\"request_buffering\":true},\"service\":{\"host\":\"petstore3.swagger.io\",\"created_at\":1698333790,\"connect_timeout\":60000,\"id\":\"069c7945-07ae-4197-a6cb-a5c1ae968d73\",\"protocol\":\"https\",\"name\":\"PetStore\",\"read_timeout\":60000,\"port\":443,\"path\":\"/api/v3\",\"updated_at\":1698399310,\"write_timeout\":60000,\"retries\":5,\"ws_id\":\"973db277-ffd7-4db0-a837-d35edc3690cc\"},\"consumer\":{\"custom_id\":\"8ac9978e8bdbe022018bdd2279f6039a\",\"created_at\":1700222094,\"id\":\"0aeebe2d-93ae-42c2-a685-2f514eb2363b\",\"tags\":null,\"username\":\"kong-demo\"}}
```
Comment (dgghinea) [inline pkg/traceability/processor/handler.go:34]: not sure if this was intended ?

File: pkg/traceability/processor/processor.go [hunk 2]
```diff
@@ -23,61 +25,61 @@ const (
...
 }
 
-func (m *EventMapper) processMapping(ctx context.Context, kongTrafficLogEntry KongTrafficLogEntry) ([]*transaction.LogEvent, error) {
-	log := log.UpdateLoggerWithContext(ctx, m.logger)
+func (m *TransactionProcessor) process() ([]beat.Event, error) {
 	centralCfg := agent.GetCentralConfig()
 	txnID := uuid.New().String()
 
-	transactionLegEvent, err := m.createTransactionEvent(kongTrafficLogEntry, txnID)
+	// leg 0
```
Comment (dgghinea) [inline pkg/traceability/processor/processor.go:51]: this one as well

**Review by dgghinea — APPROVED** [review]

File: pkg/traceability/processor/handler.go (hunk 1 above)
Comment (jcollins-axway) [reply pkg/traceability/processor/handler.go:34]: you are right, will remove it.

File: pkg/traceability/processor/processor.go (hunk 2 above)
Comment (jcollins-axway) [reply pkg/traceability/processor/processor.go:51]: this one is intentional

**Pull request merged**